src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                type(error).__name__, str(error))
+            raise Exception("Failed to calculate"
+                            " hugepage count {}".format(message))
+
+    def edit_dell_environment_file(
+            self,
+            enable_hugepage,
+            enable_numa,
+            hugepage_size,
+            hostos_cpu_count,
+            dell_compute_count=0):
+        try:
+            logger.info("Editing dell environment file")
+            file_path = home_dir + '/pilot/templates/dell-environment.yaml'
+            if not os.path.isfile(file_path):
+                raise Exception(
+                    "The dell-environment.yaml file does not exist")
+            cmds = ['sed -i "s|  # NovaSchedulerDefaultFilters|  ' +
+                    'NovaSchedulerDefaultFilters|" ' + file_path]
+            cmds.append(
+                'sed -i "s|dellnfv::hugepages::enable:.*'
+                '|dellnfv::hugepages::enable: ' +
+                str(enable_hugepage) +
+                '|" ' +
+                file_path)
+            cmds.append(
+                'sed -i "s|dellnfv::numa::enable:.*|dellnfv::numa::enable: ' +
+                str(enable_numa) +
+                '|" ' +
+                file_path)
+
+            cmds.append(
+                'sed -i "s|DellComputeCount:.*|DellComputeCount: ' +
+                str(dell_compute_count) +
+                '|" ' +
+                file_path)
+
+            if enable_hugepage:
+                hugepage_number = ConfigOvercloud.calculate_hugepage_count(
+                    hugepage_size)
+                cmds.append('sed -i "s|dellnfv::hugepages::hugepagesize:.*'
+                            '|dellnfv::hugepages::hugepagesize: ' +
+                            hugepage_size[0:-1] + '|" ' + file_path)
+                cmds.append(
+                    'sed -i "s|dellnfv::hugepages::hugepagecount:.*'
+                    '|dellnfv::hugepages::hugepagecount: ' +
+                    str(hugepage_number) +
+                    '|" ' +
+                    file_path)
+            if enable_numa:
+                vcpu_pin_set = ConfigOvercloud.calculate_hostos_cpus(
+                    hostos_cpu_count)
+                cmds.append(
+                    "sed -i 's|dellnfv::numa::vcpu_pin_set:.*"
+                    "|dellnfv::numa::vcpu_pin_set: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+                cmds.append(
+                    'sed -i "s|  # NovaVcpuPinSet|  ' +
+                    'NovaVcpuPinSet|" ' +
+                    file_path)
+                cmds.append(
+                    "sed -i 's|NovaVcpuPinSet:.*|NovaVcpuPinSet: \"" +
+                    vcpu_pin_set +
+                    "\"|' " +
+                    file_path)
+
+            for cmd in cmds:
+                status = os.system(cmd)
+                logger.debug("cmd: {}".format(cmd))
+            if status != 0:
```

Comment:
I think the above line and the below raise statement need to be indented so they are inside the for loop, otherwise only a failure of the last command executed will be detected.

Reply:
Aggred
Moved in loop